### ISSUE TITLE
unsigned and empty notes can be added through patient profile without alert 

### ISSUE THREAD

**Issue by MostafaHamedAbdelmasoud** [LibreHealthIO/lh-ehr#1403]
if you click on (patient profile) -> then on ( notes), it is gonna expand and show you notes, then try to add epmpty and unsigned note, it will be added, so i think it's not acceptable here and should be improved

![screenshot 94](https://user-images.githubusercontent.com/36406025/52289977-c7b6be00-2977-11e9-8ef6-b0af0067a800.png)

**Comment by aethelwulffe** [comment]
Yes, the Patient Notes should be looked at just as you are doing so.  There is also some more concern about someone being able to inactivate a message in there.  These are messages about the patient, and not like clinical intervention notes, but they are part of the patient chart so to speak.  Some of it is ephemeral information (temporary) but that does not mean the record of entries need ever be dropped (unless it needed to be moved to the correct patient!).

@MostafaHamedAbdelmasoud We should work on this, yes.  I think before diving into it too deeply, we should get the input of a particular heavy user of the patient notes tool.  Their clinic has been using it for about 15 years.  They might use it overly-heavy as a matter of fact, in lieu of a better tool.
  I will contact Dr. Sam Bowen, and see if I can get his immediate input, but he has given deep rants on this subject.
Without his permission, I am going to post some totally out-of context excerpts from emails and discussions about the pnotes feature and the issues withing it.  This for purposes of mining out hints on things that are problematic:

> Do not delete messages.  Do not mark them inactive.  Do not mark them 'Message Status'='Done'.
> 
> I routinely go through the database and change all messages to Delete = false, Activity=true, 'Massage Status'='New'
> 
> This upsets everyone when I do this because all of their messages suddenly pop back up.  It means you have not been complying with previous requests not to click the "Done" message status.
> 
> Marking messages in any way that removes from the message log is not a good medical practice.    I have been strongly urging all users not mark 'message status'='Done' sine we started using the OpenEMR since 2004.  Using the 'Message Status'=Done hides the message from the message log.    Please get into the habit of sending all finished messages to user "Done, I am".
> 
> This has nothing to do with the "Dated Messages" as discussed previously.
> 
> Sam Bowen, MD 

You will note in the above that the original developer used labels like "Massage Status".  Not great Q.A.
The second item is that they have been using this since before the "Dated Reminders" system.  They use it as an Instant Message service for communications regarding individual patients.

The ideas on the message system within the ehr are worth discussing.  

For patient Notes specifically, there needs to be some access control on who can do what as far as changing the status of a message.  That means it needs to be "resolved to me" and not "resolved globally" like the Dated Reminders can do.  This is in line with what you are reporting here.

In context to communications within the EHR, I think the following things should be considered in the context of replacing Patient Notes and Dated Reminders.

Integrated communication tool, but should use separate tables for office communications vs. Patient Related office communications.  Context is important there. 

- Proper InMail type tool that also serves as a Reminder system.
- Patient-Specific Notes (Chart Notes...not the same as Clinical Forms or Encounters).
- Links in the InMail system that allow a touchlist for Patient Notes, but have the final firewall of Access Control to ensure that any observed link that is not within the recipient's access level simply does not work.   This requires reminders that are patient-relevant to be dependent on the InMail tool, but ensures that Access Control only has to be called at a single point.
- Possibly an informal communication IM with basic features.

**Comment by tafodinho** [comment]
@MostafaHamedAbdelmasoud I will like to work on this issue as well

**Comment by muarachmann** [comment]
@MostafaHamedAbdelmasoud are you willing to work on this if yes go ahead. If no @tafodinho you can work on it

**Comment by muarachmann** [comment]
@tafodinho please look for another issue to work on. Its been already assigned

**Comment by tafodinho** [comment]
@muarachmann ok

**Comment by muarachmann** [comment]
> @muarachmann ok

@tafodinho you might want to look at #1400
